### ISSUE TITLE
[1.15] objectives show "%d turns left" in non en_US locale

### ISSUE THREAD

**Issue by stevecotton** [wesnoth/wesnoth#4165]
In 1.15, the scenario objectives of any scenario that uses `{TURNS_RUN_OUT}` will show a `%d` to the user. The bug is locale-specific:

* in en_US, the bug doesn't happen. The user sees "Turns run out (24 turns left)"
* in en_GB, the user sees "Turns run out (%d turns left)"
* in Spanish, the user sees "Se acaban los turnos (%d turnos restantes)

I haven't debugged why, but 12ada04b8f changed the plural form to "($remaining_turns turns left)" without changing the singular form. I'm guessing Gettext is using the translation with msgid "(this turn left)" even though the msgid_plural doesn't match.

Platform: Debian Unstable, amd64
Boost version: 1.67
Building Wesnoth from commit: 0bd3682b664
Reproduced in scenarios: AoI S1, WoV S2

**Comment by jostephd** [comment]
> I'm guessing Gettext is using the translation with msgid "(this turn left)" even though the msgid_plural doesn't match.

That does seem to be what's happening; you can see it easily by changing the plural strings in HttT S23: https://github.com/wesnoth/wesnoth/blob/ead1e7a3937bac34e8247b271628cfbfd5790c1e/data/campaigns/Heir_To_The_Throne/scenarios/23_Test_of_the_Clans.cfg#L19-L37

This does beg the question of what string freeze for 1.15.0 will start. @Vultraz ?

**Comment by Vultraz** [comment]
What needs to be done to fix this, then?

**Comment by jostephd** [comment]
Not sure. Maybe regenerating the po files will fix it? But then we'll have an untranslated plural string.

**Comment by AI0867** [comment]
This is something that can be fixed mechanically: replacing `%d` with `$remaining_turns|` in the `msgid_plural` and the `msgstr`s should do the trick. Can pofix do this?

**Comment by jostephd** [comment]
@AI0867 I guess so but I'm not sure. Also, that would fix the problem only for this one string. Are there any other strings where the plural message was changed but the singular message not changed?

**Comment by gfgtdf** [comment]
> This does beg the question of when string freeze for 1.15.0 will start. @Vultraz ?

Iirc Dev releases have no string freeze, but I could be wrong.

**Comment by CelticMinstrel** [comment]
I think pofix can do it only if there's no other strings that use `%d`.

**Comment by jostephd** [comment]
@gfgtdf I assumed we'd want to have uptodate translations in 1.15.0 to encourage testing by players who don't speak English. I guess most of the UI strings and so on will carry over from 1.14, but areas that have new strings are generally exactly the areas we want to have testing of.

**Comment by soliton-** [comment]
Having uptodate translations certainly doesn't hurt but it also means lots of work for translators that can be in vain at any moment because we decide to reorganize/rewrite stuff. Usually translators get more active when we're nearing a new stable version.

**Comment by Wedge009** [comment]
Is this still an issue? I don't see this problem with current 1.15.1+dev, using en_GB or fr.

**Comment by stevecotton** [comment]
> Are there any other strings where the plural message was changed but the singular message not changed?

Checking master at b4c65f179 :

For the "turns left" message, there are many commented-out fuzzy strings with %d markers, they might resurface with the %d still in them. In the languages that have different plurals for 2, 3 or 4 items, there's a lot of commented-out fuzzies where all of the plurals are exactly the same string (with no number in it).

There's a similar error for the gold carry-over in a lot of languages (cs, el, en_GB, es, et, fi, fr, gd, gl, he, hu, id, it, ja, ko, la, lt, lv, nl, pl, pt, pt_BR, ru, sk, sr, sr@ijekavian, sr@ijekavianlatin, sr@latin, sv, tr, uk, vi, zh_CN, zh_TW)
```
po/wesnoth/de.po-msgid "$percent|% of gold carried over to the next scenario."
po/wesnoth/de.po:msgstr "%d%% Goldstücke werden ins nächste Szenario übernommen."
```
When playing in German, that string shows shows up in English instead of German in the game, with the percentage correctly substituted.

**Comment by stevecotton** [comment]
> Is this still an issue? I don't see this problem with current 1.15.1+dev, using en_GB or fr.

Seems somebody's re-organised the .lua files, and some of them are missing `-- #textdomain` comments. I expect to open a PR that changes the fuzzy strings back to non-fuzzy in the next hour or so.

**Comment by jostephd** [comment]
> Edit: they don't need textdomain comments (as it defaults to "wesnoth"), the problem is #4275 .

Can we work around that with something like

```
if false
    _ "(this turn left)"
    _ "($remaining_turns turns left)"
end
```

to make wmlxgettext find the strings, without affecting runtime?

**Comment by CelticMinstrel** [comment]
That's wrong. This is a plural string - what you suggest would make wmlxgettext detect it as two singular strings.

**Comment by stevecotton** [comment]
Fixes for #4275 have now been applied to both 1.14 and 1.15. For this issue, the remaining tasks seem to be:
* (optional) fix the obvious ones that don't need knowledge of the target language
* close this issue now, the remaining instances will be fixed by the translators

**Comment by jostephd** [comment]
> close this issue now, the remaining instances will be fixed by the translators

Sounds like we should collect somewhere the stuff to tell translators in the next announcement: there's also https://github.com/wesnoth/wesnoth/pull/4390#issuecomment-536295526.
